mkosi/backend.py

```diff
@@ -492,15 +492,18 @@ def safe_tar_extract(tar: tarfile.TarFile, path: Path=Path("."), *, numeric_owne
     See https://github.com/advisories/GHSA-gw9q-c7gh-j9vm
     """
     path = path.resolve()
+    members = []
     for member in tar.getmembers():
         target = path / member.name
         try:
             # a.relative_to(b) throws a ValueError if a is not a subpath of b
-            target.resolve().relative_to(path)
+            if not (member.ischr() or member.isblk()):
```

Comment:
Please document this change and put it in a separate commit

Reply:
this is the result of dropping root privileges. should it be included in the commit?

Reply:
This is a self-contained change on its own, where we don't unpack character or device nodes from an archive. So it makes sense on its own and should be a separate commit.